### PR TITLE
update dependency notices for 6.6.1 release

### PR DESCRIPTION
**Pull request by yaauie**
dependency report runs green with this

### CODE REVIEW

**Comment by robbavey** [comment]
@yaauie Do you know why the `NOTICES.txt` file has changed so significantly? And the importance of those changes?

There are some licenses specified in the `csv` file that are no longer in the `NOTICES` file, such as `asm`, `backports`, and the `jnr` libraries. However, as far as I can they are still in the `tools/dependencies-report/src/main/resources/notices/` folder.

**Comment by yaauie** [comment]
> Do you know why the `NOTICES.txt` file has changed so significantly? And the importance of those changes?

It looks like we haven't committed changes to `NOTICES.txt` since it was improved in 62d00da, which changed the format a bit. That coupled with changes in dependencies themselves in that timeframe, and we have a pretty big diff.

> There are some licenses specified in the `csv` file that are no longer in the `NOTICES` file, such as `asm`, `backports`, and the `jnr` libraries. However, as far as I can they are still in the `tools/dependencies-report/src/main/resources/notices/` folder.

I'll take a look and see what I can do to cleanup.

**Comment by yaauie** [comment]
@robbavey I have audited to ensure:
 -  we aren't including things in the NOTICE that don't need to be there,
 -  that the csv no longer specifies things that aren't runtime dependencies, and
 - that the notices matching dependencies that were removed from the csv are also deleted.

I have also done a partial-revert of the months-old refactor in the notice generation that changed the format of the dependency headers, since the prior format was more useful and matches the committed `/NOTICE.TXT`. I plan to open a PR to port this change forward to `master`, `7.x`, and `7.0`, along with a similar audit of their notices.

Changes in `/NOTICE.TXT` are _slightly_ easier to make sense of with `git show --minimal <rev>`, which is less likely to get confused by newlines.

File: tools/dependencies-report/src/main/resources/notices/dalli-2.7.9-NOTICE.txt [hunk 1]
```diff
@@ -1,4 +1,6 @@
-Copyright (c) Jim Weirich
```
Comment (yaauie) [inline tools/dependencies-report/src/main/resources/notices/dalli-2.7.9-NOTICE.txt]: it looks like git is trying a little _too_ hard to find file renames.

**Pull request merged**